Rock/Web/UI/Controls/SearchField.cs

```diff
@@ -58,13 +60,25 @@ public override void RenderControl( HtmlTextWriter writer )
             hfFilter.ID = this.ID + "_hSearchFilter";
 
             var searchExtensions = new Dictionary<string,Tuple<string, string>>();
-            foreach ( KeyValuePair<int, Lazy<Rock.Search.SearchComponent, Rock.Extension.IComponentData>> service in Rock.Search.SearchContainer.Instance.Components )
+
+            var rockPage = this.Page as RockPage;
+            if ( rockPage != null )
             {
-                if ( !service.Value.Value.AttributeValues.ContainsKey( "Active" ) || bool.Parse( service.Value.Value.AttributeValues["Active"].Value ) )
+                var currentPerson = rockPage.CurrentPerson;
+                if ( currentPerson != null )
                 {
-                    searchExtensions.Add( service.Key.ToString(), Tuple.Create<string, string>( service.Value.Value.SearchLabel, service.Value.Value.ResultUrl ) );
-                    if ( string.IsNullOrWhiteSpace( hfFilter.Value ) )
-                        hfFilter.Value = service.Key.ToString();
+                    foreach ( KeyValuePair<int, Lazy<Rock.Search.SearchComponent, Rock.Extension.IComponentData>> service in Rock.Search.SearchContainer.Instance.Components )
```

Comment:
Security may be configured to allow access even if current person is null (i.e. All Users). Instead of checking for a valid currentPerson, just pass the rockPage.CurrentPerson value to the IsAuthorized method. It will handle the null value.